src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -287,37 +330,54 @@ const StudentApplication = ({ userProfile, authentication }) => {
    *
    * @async
    * @function changeApplicationEditor
-   * @param {String} newEditorUsername the student username of the person who will be allowed to edit this application
+   * @param {StudentProfileInfo} newEditorProfile the StudentProfileInfo object for the person who will be allowed to edit this application
    */
-  const changeApplicationEditor = async (newEditorUsername) => {
+  const changeApplicationEditor = async (newEditorProfile) => {
     setSaving(true);
     setSaveButtonAlertTimeout(null);
     try {
       const result = await housing.changeApartmentAppEditor(
         applicationDetails.ApplicationID,
-        newEditorUsername,
+        newEditorProfile.AD_Username,
       );
-      console.log(result); //! DEBUG
-      setApplicationDetails((prevApplicationDetails) => ({
-        ...prevApplicationDetails,
-        EditorProfile: newEditorProfile,
-      }));
-      setSaving('success');
-      setUnsavedChanges(true);
-      // loadApplication(); //? Coming soon to a feature near you
-    } catch {
-      setSnackbarText('Something went wrong while trying to save the new application editor.');
-      setSnackbarSeverity('error');
-      setSnackbarOpen(true);
-      setSaving('failed');
+      if (result) {
+        try {
+          const result = loadApplication();
+          if (!result) {
```

Comment:
It's a little bit confusing to shadow the `result` variable declared in the outer `try-catch` block by redeclaring `result` in this inner `try-catch` block. Could you use a different name to distinguish them?

Reply:
Yes, those should be a different name

Reply:
Yes, those should be a different name